### PR TITLE
Added scope in_week + tests

### PR DESCRIPTION
**Pull request by jtkpiotr**
@mhib @quirk0o @lkudyba @adamzima

### CODE REVIEW

File: app/models/event.rb [hunk 1]
```diff
@@ -9,4 +9,11 @@ def start_time_must_be_lower_than_end_time
     errors.add(:start_time, "Can't be higher than end_time") if start_time.try(:>, end_time)
   end
 
+  scope :in_week, ->(week) {
+    query = '(start_time >= :start_week AND start_time <= :end_week)'\
+    'OR (end_time >= :start_week AND end_time <= :end_week )'\
+    'OR (start_time <= :start_week AND end_time >= :end_week)'
+    Event.where(query, start_week: week.beginning_of_week, end_week: week.end_of_week)
```
Comment (pbzymek) [inline app/models/event.rb]: here you shoudn't use Event as where method is available:
`where(query, start_week: week.beginning_of_week, end_week: week.end_of_week)`

Comment (pbzymek) [inline app/models/event.rb]: you might also want to look into this gem: https://github.com/jeremyevans/sequel

File: spec/models/event_spec.rb [hunk 2]
```diff
@@ -17,4 +17,25 @@
       expect(event).to be_valid
     end
   end
+
+  describe '#in_week' do
+    let(:start_time) { Time.now.beginning_of_week }
+    let!(:events) {
```
Comment (pbzymek) [inline spec/models/event_spec.rb]: how about dividing the let content into two:
`let!(:expected_events) { ... }`
`let!(:other_event) { ... }`

File: spec/models/event_spec.rb [hunk 3]
```diff
@@ -17,4 +17,25 @@
...
+          FactoryGirl.create(:event, start_time: start_time - 2.days, end_time: start_time - 2.days + 1.hours),
+          FactoryGirl.create(:event, start_time: start_time - 2.days, end_time: start_time + 2.days),
+          FactoryGirl.create(:event, start_time: start_time - 2.days, end_time: start_time + 10.days),
+          FactoryGirl.create(:event, start_time: start_time + 1.days, end_time: start_time + 10.days)
+
+      ]
+    }
+
+    it "returns all events from specified week" do
+      expected_events = events.clone.tap { |n| n.delete_at(2) }
```
Comment (pbzymek) [inline spec/models/event_spec.rb]: don't do clone and then sort the collection just to compare them, here's a few matchers for collections:
`expect(actual).to include(expected)
expect(array).to match_array(expected_array)
# ...which is the same as:
expect(array).to contain_exactly(individual, elements)`

File: spec/models/event_spec.rb [hunk 4]
```diff
@@ -17,4 +17,27 @@
...
+      ]
+    }
+    let!(:not_expected_events) {
+      [
+          FactoryGirl.create(:event, start_time: start_time - 2.days, end_time: start_time - 2.days + 1.hours)
+      ]
+    }
+
+    it "returns all events from specified week" do
+      expect(Event.in_week(start_time)).to match_array expected_events
```
Comment (pbzymek) [inline spec/models/event_spec.rb]: out of curiosity, does the order in this matter?

File: app/models/event.rb [hunk 5]
```diff
@@ -9,4 +9,11 @@ def start_time_must_be_lower_than_end_time
     errors.add(:start_time, "Can't be higher than end_time") if start_time.try(:>, end_time)
   end
 
+  scope :in_week, ->(week) {
+    query = '(start_time >= :start_week AND start_time <= :end_week)'\
+    'OR (end_time >= :start_week AND end_time <= :end_week )'\
+    'OR (start_time <= :start_week AND end_time >= :end_week)'
+    where(query, start_week: week.beginning_of_week, end_week: week.end_of_week)
```
Comment (adamzima) [inline app/models/event.rb]: - for ranges use `where(start_time: (week.beginning_of_week..week.end_of_week))`
- for or consider using cutting edge `or` relations - https://github.com/rails/rails/blob/master/activerecord/test/cases/relation/or_test.rb

File: spec/models/event_spec.rb (hunk 4 above)
Comment (adamzima) [inline spec/models/event_spec.rb]: use `described_class` instead of `Event` explicitely

File: spec/requests/calendars_spec.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe "Calendars", type: :request do
```
Comment (adamzima) [inline spec/requests/calendars_spec.rb]: :+1:

File: spec/models/event_spec.rb [hunk 7]
```diff
@@ -17,4 +17,27 @@
...
     end
   end
+
+  describe '#in_week' do
+    let(:start_time) { Time.now.beginning_of_week }
+    let!(:expected_events) {
+      [
+          FactoryGirl.create(:event, start_time: start_time - 2.days, end_time: start_time + 2.days),
+          FactoryGirl.create(:event, start_time: start_time - 2.days, end_time: start_time + 10.days),
+          FactoryGirl.create(:event, start_time: start_time + 1.days, end_time: start_time + 10.days),
```
Comment (adamzima) [inline spec/models/event_spec.rb]: Uo these are really long events :D

File: app/models/event.rb (hunk 5 above)
Comment (s1mplex) [inline app/models/event.rb]: I think you can simplify this condition to:

`:end_week >= start_time AND :start_week <= end_time`

**Pull request merged**